lib/collection/src/collection/payload_index_schema.rs

```diff
@@ -84,4 +84,37 @@ impl Collection {
 
         Ok(result)
     }
+
+    /// Returns an arbitrary payload key used by `filter` which can be indexed but currently is not.
+    /// If this function returns `None` all indexable keys in `filter` are indexed.
+    pub fn has_filter_without_index(&self, filter: &Filter) -> Option<JsonPath> {
+        filter
+            .must
+            .iter()
+            .flatten()
+            .chain(filter.must_not.iter().flatten())
+            .chain(filter.should.iter().flatten())
+            .chain(filter.min_should.iter().flat_map(|i| &i.conditions))
+            .find_map(|condition| self.has_condition_without_index(condition))
+    }
+
+    fn has_condition_without_index(&self, condition: &Condition) -> Option<JsonPath> {
```

Comment:
This function doesn't see to take index type into account.

E.g. if index is built for integer and query uses keyword